content/root/best-practice-guide.textile

```diff
@@ -5,45 +5,83 @@ index: 30
+h3(#ensuring-tokens-are-auto-renewable). Renewing tokens
+
+Tokens are short-lived, that is, they expire after a given time. You must ensure that the client can request a new token when it needs one. The Ably Client Library SDKs provide two different mechanisms to obtain tokens automatically: "@authUrl@":/realtime/authentication#token-upgrading or "@authCallback@":/realtime/authentication#token-upgrading. You specify which one to use in the "ClientOptions":/realtime/authentication#token-authentication object.
 
 h3(#rest-or-realtime). Realtime vs REST
 
-Ably provides two types of client libraries. "Realtime":/realtime libraries maintain a persistent connection to Ably and can be considered stateful, whereas "REST":/rest libraries issue HTTP requests to the Ably platform and are stateless.
+Ably Client Library SDKs may implement two interfaces: realtime and REST. The "Realtime":/realtime interface maintains a persistent connection to Ably and can be considered stateful. The "REST":/rest interface issues HTTP requests to the Ably platform and is stateless. The REST interface is therefore lighter weight because it does not need to maintain state. 
+
+Depending on your use case, you may only need to use the REST interface on your server. For example, the server may only need to authenticate with Ably, and then act as an auth server for clients. You might then use the realtime interface on your client, so it can both subscribe and publish messages.
+
+The PHP and Python SDKs only have a REST interface. While the Ruby SDK implements both realtime and REST interfaces, there is also a Ruby SDK available that implements only the REST interface. The full list of available SDKs, and their supported interfaces, can be found on the "SDK download":https://ably.com/download page.
+
+An important difference between the REST and realtime interface is that with REST you can publish but not subscribe to realtime messages. With the realtime interface you can both publish and subscribe. If you have a use case where you only need to authenticate, or publish messages, then the REST interface is recommended.
+
+The main factors are summarized here:
+
+* If you want to maintain a persistent connection with Ably's Realtime platform in order to subscribe to messages sent on a channel, or be notified whenever the presence set changes, you need to use the "realtime library":/realtime.
+* If you just want to just publish data to the platform and don't need to subscribe or manage channel attachment, you can the "REST library":/rest.
+* If you want to publish at very high rates or with the lowest possible realtime latencies, "Ably's realtime library would be best as it can publish at higher rates and lower latencies than is possible with the REST library due to message pipelining":https://knowledge.ably.com/should-i-use-the-rest-or-realtime-library.
```

Comment:
Would it be clearer to say that this is possible because the realtime library sends these messages over a websocket, rather than talk about pipelining?

Reply:
Yes, I think it would. I will capture that in another ticket and PR though, as I'd like to do a little extra research first before rewriting that section.